include/nccl_ofi_memcheck_asan.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020-2023 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ */
+
+#ifndef NCCL_OFI_MEMCHECK_ASAN_H
+#define NCCL_OFI_MEMCHECK_ASAN_H
+
+#ifdef _cplusplus
+extern "C" {
+#endif
+
+#if !defined(__SANITIZE_ADDRESS__)
+#error "memcheck-asan should not be compiled when ASAN is disabled"
+#endif
+
+/*
+ * Marks memory region [addr, addr+size) as unaddressable.
```

Comment:
shouldn't you get these from `sanitizer/asan_interface.h`?

Reply:
I will remove the declarations and include the header in the next revision.